src/core/plugins/core-hud/client/index.ts

```diff
@@ -272,6 +274,18 @@ class InternalFunctions implements ViewModel {
     static defaultInteractionsComponent(propName) {
         InternalFunctions.passComponentInfo(propName, JSON.stringify(interactions.concat(customInteractions)), true);
     }
+
+
+    static defaultStreetComponent(propName: string) {
+        const streetHash = native.getStreetNameAtCoord(alt.Player.local.pos.x, alt.Player.local.pos.y, alt.Player.local.pos.z, null, null,)[1];
+        const value = native.getStreetNameFromHashKey(streetHash)
```

Comment:
Store the players current position rather than getting it 3 times - it's much better performance wise

```ts
const pos = alt.Player.loca.pos;
const streetHash = native.getStreetNameAtCoord(pos.x, pos.y, pos.z, null, null,)[1];
```